helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt

```diff
@@ -0,0 +1,187 @@
+import org.ossreviewtoolkit.model.config.OrtConfiguration
+import org.ossreviewtoolkit.model.config.PostgresStorageConfiguration
+import org.ossreviewtoolkit.model.jsonMapper
+import org.ossreviewtoolkit.model.utils.DatabaseUtils
+import org.ossreviewtoolkit.utils.common.expandTilde
+import org.ossreviewtoolkit.utils.core.Environment
+import org.ossreviewtoolkit.utils.core.ORT_CONFIG_FILENAME
+import org.ossreviewtoolkit.utils.core.ortConfigDirectory
+
+internal class CreateAnalyzerResultCommand : CliktCommand(
+    help = "Creates an analyzer result that contains packages for the given list of package ids. The result contains " +
+            "only packages which have a corresponding result in the postgres storage."
+) {
+    private val packageIdsFile by option(
+        "--package-ids-file",
+        help = "The list of package ids to put into the output analyzer result in plain text with one entry per line."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val ortFile by option(
+        "--ort-file", "-o",
+        help = "The ORT file to write the generated synthetic analyzer result to."
+    ).convert { it.expandTilde() }
+        .file(mustExist = false, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val configFile by option(
+        "--config",
+        help = "The path to the ORT configuration file that configures the scan results storage."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .default(ortConfigDirectory.resolve(ORT_CONFIG_FILENAME))
+
+    private val configArguments by option(
+        "-P",
+        help = "Override a key-value pair in the configuration file. For example: " +
+                "-P ort.scanner.storages.postgres.schema=testSchema"
+    ).associate()
+
+    override fun run() {
+        val ids = packageIdsFile.readLines().filterNot { it.isBlank() }.map { Identifier(it.trim()) }
+        val packages = openDatabaseConnection().use { getScannedPackages(it, ids) }.filterMaxByDate()
+        val ortResult = createAnalyzerResult(packages)
+
+        println("Writing analyzer result with ${packages.size} packages to '${ortFile.absolutePath}'.")
+        writeOrtResult(ortResult, ortFile)
+    }
+
+    private fun openDatabaseConnection(): Connection {
+        val storageConfig = OrtConfiguration.load(configArguments, configFile).scanner.storages
+            ?.get("postgres") as? PostgresStorageConfiguration
+            ?: throw IllegalArgumentException("postgresStorage not configured.")
```

Comment:
As the names in the configuration are arbitrary you could make this more generic with:
```
config.storages?.values?.filterIsInstance<PostgresStorageConfiguration>()?.firstOrNull() ?:  throw IllegalArgumentException("...")
```